codespace: choose codespace prompt improvements

josebalius opened the pull request:
This PR changes the way the choose codespace prompt works. It prioritizes rendering the codespaces by `repo: branch` and only includes the codespace name where is a name conflict of multiple codespaces on the same repo/branch.

For example:
```
❯ ghd cs logs
? Choose codespace:  [Use arrows to move, type to filter]
> josebalius/docs: jg/error-handler-docs
  github/github: master*
  josebalius/fiber: jg/mounted-err-handler-support
  josebalius/dotfiles: main [josebalius-dotfiles-cppafda43q24ijafls]
  cli/cli: trunk
  josebalius/dotfiles: main [josebalius-dotfiles-9vvj57hp96r]
```

As part of this PR I moved the `api.Codespace` declaration into its own pkg `codespace` as `codespace.Codespace` to contain and group all behavior associated with the codespace model. This includes:

- Checking for unsaved changes
- Render the branch with git status
- Display name logic

Some of this code was duplicated and/or would need to be further duplicated as part of this change so this is an improvement I was looking forward to.

I hope to further clean up the API pkg by removing the `User` and `Repository` declarations to re-use existing `gh` values but I will do that in a different PR.

josebalius committed:
Change choose codespace prompt

- Repo + branch is favored
- Codespace name is included to disambiguate between two codespaces
- Move Codespace model out of out API into its own package
- Update call sites and group behavior under codespace.Codespace

josebalius committed:
Merge branch 'trunk' of github.com:cli/cli into jg/choose-codespace-prompt

josebalius committed:
Fixes bug there are +2 codespaces with a conflict

- Tracks conflicting name going forward for other records
- Moves the git status dirty star into a constant so we can reference it

josebalius committed:
Better short name

josebalius committed:
Merge branch 'trunk' of github.com:cli/cli into jg/choose-codespace-prompt

josebalius commented:
I am going to be updating this PR a bit to postpone the API changes and instead wrap the `*api.Codespace` struct within the `cmd/codespace` pkg.

mislav reviewed (approved):
The new selector looks great!

As discussed, a potential improvement to this would be to organize things differently so there is not an awkwardly-named package `internal/codespaces/codespace`, but other than that, this is good to go 👍

josebalius committed:
Keep codespace struct in API for now

- Use a private codespace structure in the cmd pkg to encapsulate common
  behavior

josebalius committed:
Remove unused guid

josebalius committed:
Update docs

mislav reviewed (approved):
Looks great!

A lot of variable names are now (seemingly unnecessarily) renamed in this PR; if you wanted to slim down the diff further then those can be reverted.

josebalius committed:
Merge branch 'trunk' of github.com:cli/cli into jg/choose-codespace-prompt

josebalius committed:
Revert other rename changes

josebalius committed:
Revert rename for ports cmd
